fili-core/src/main/java/com/yahoo/bard/webservice/web/ApiFilter.java

```diff
@@ -172,6 +173,26 @@ public FilterOperation getOperation() {
         return this.values;
     }
 
+    /**
+     * Take two Api filters which differ only by value sets and union their value sets.
+     *
+     * @param one  The first ApiFilter
+     * @param two  The second ApiFilter
+     *
+     * @return an ApiFilter with the union of values
+     */
+    public static ApiFilter union(ApiFilter one, ApiFilter two) {
+        if (!Objects.equals(one.getDimension(), two.getDimension())
+                && Objects.equals(one.getDimensionField(), two.getDimensionField())
+                && Objects.equals(one.getOperation(), two.getOperation())
```

Comment:
The first '!' is incorrect, it should have been a not distributed over the whole boolean.